### PR TITLE
Cached languages

### LINKED ISSUE #2344: Cache languages button content in LoadLanguages
Just noticed when debugging that LoadLanguages gets called each time loading a save or entering the game. I think using a lazy initialized static list to hold the locale data that will go into the OptionButton would be better. Should be relatively easy to take the approach in Constants.cs that guarantees lazy initialization without using the Lazy class.

### PR DESCRIPTION
**Pull request by Kemikal1**
**Brief Description of What This PR Does**

closes #2344

**Related Issues**

**Progress Checklist**

Note: before starting this checklist the issue should be marked as non-draft.

- [x] PR author has checked that this PR works as intended and doesn't
      break existing features:
      https://wiki.revolutionarygamesstudio.com/wiki/Testing_Checklist
      (this is important as to not waste the time of Thrive team
      members reviewing this PR)
- [ ] Initial code review passed
- [ ] Functionality is confirmed working by another person (see above checklist link)
- [ ] Final code review is passed and code conforms to the 
      [styleguide](https://github.com/Revolutionary-Games/Thrive/blob/master/doc/style_guide.md).

Before merging all CI jobs should finish on this PR without errors, if
there are automatically detected style issues they should be fixed by
the PR author.

### CODE REVIEW

**Comment by Kemikal1** [comment]
Files kept changing for some reason :/ so i made a new pull request.

File: src/general/OptionsMenu.cs [hunk 1]
```diff
@@ -162,6 +162,11 @@ public class OptionsMenu : Control
     [Export]
     public NodePath CustomUsernamePath;
 
+    // Sound tab statics
+    private List<string> languages = TranslationServer.GetLoadedLocales().Cast<string>()
+        .OrderBy(i => i, StringComparer.InvariantCulture)
+        .ToList();
```
Comment (hhyyrylainen) [inline src/general/OptionsMenu.cs:168]: This needs to be static to actually cache things...
For that you need to take a look at how the game version is loaded in Constants.cs to see how the lazy load is implemented there.

**Review by hhyyrylainen — CHANGES_REQUESTED** [review]

**Review by FuJa0815 — APPROVED** [review]
Have not experianced any ingame bugs

File: src/general/OptionsMenu.cs [hunk 2]
```diff
@@ -162,6 +162,11 @@ public class OptionsMenu : Control
     [Export]
     public NodePath CustomUsernamePath;
 
+    // Sound tab statics
+    private static List<string> languages = TranslationServer.GetLoadedLocales().Cast<string>()
```
Comment (hhyyrylainen) [inline src/general/OptionsMenu.cs:166]: This triggers too soon (before the first actual usage). Still please look at how Constants.cs does it:
https://github.com/Revolutionary-Games/Thrive/blob/cf56b89a03d8dbf5067a260d14e319170fc6d1b4/simulation_parameters/Constants.cs#L544
https://github.com/Revolutionary-Games/Thrive/blob/cf56b89a03d8dbf5067a260d14e319170fc6d1b4/simulation_parameters/Constants.cs#L539

Doing it like that *only* causes the static variable to initialize when the property is *first* read, and not during static variable initialization phase.

**Review by hhyyrylainen — CHANGES_REQUESTED** [review]

File: src/general/OptionsMenu.cs (hunk 2 above)
Comment (FuJa0815) [reply src/general/OptionsMenu.cs:166]: I'd write something like this to make sure the languages only get fetched when required:
```cs
private List<string> languages;
private List<string> Languages => languages ??= FetchLanguages();
private List<string> FetchLanguages() {
    return TranslationServer.GetLoadedLocales().Cast<string>()
        .OrderBy(i => i, StringComparer.InvariantCulture)
        .ToList();
}
```

Comment (hhyyrylainen) [reply src/general/OptionsMenu.cs:166]: That's not thread safe. See my example in Constants.cs
I looked this into a lot when I wrote that, and that is the *only* thread safe single initialization on demand for a variable without using Lazy I could find.

File: src/general/OptionsMenu.cs [hunk 3]
```diff
@@ -162,6 +162,11 @@ public class OptionsMenu : Control
     [Export]
     public NodePath CustomUsernamePath;
 
+    // Sound tab statics
+    private static readonly List<string> Languages_N = TranslationServer.GetLoadedLocales().Cast<string>()
```
Comment (hhyyrylainen) [inline src/general/OptionsMenu.cs:166]: This doesn't follow the naming convention in our style guide.

**Review by hhyyrylainen — CHANGES_REQUESTED** [review]

File: src/general/OptionsMenu.cs [hunk 4]
```diff
@@ -162,6 +162,11 @@ public class OptionsMenu : Control
     [Export]
     public NodePath CustomUsernamePath;
 
+    // Sound tab statics
```
Comment (hhyyrylainen) [inline src/general/OptionsMenu.cs:165]: Isn't this comment wrong?

File: src/general/OptionsMenu.cs [hunk 5]
```diff
@@ -162,6 +162,11 @@ public class OptionsMenu : Control
     [Export]
     public NodePath CustomUsernamePath;
 
+    // Sound tab cache
```
Comment (hhyyrylainen) [inline src/general/OptionsMenu.cs:165]: I think this comment should just be removed, as there's just one variable here. If you want you could add an xml doc comment for this variable.

Comment (Kemikal1) [reply src/general/OptionsMenu.cs]: Shoulde said so from the beginning........

File: src/general/OptionsMenu.cs [hunk 6]
```diff
@@ -265,6 +268,9 @@ private enum SelectedOptionsTab
         Miscellaneous,
     }
 
+    // Languages variable
```
Comment (hhyyrylainen) [inline src/general/OptionsMenu.cs:271]: This comment is pretty much useless...

Comment (Kemikal1) [reply src/general/OptionsMenu.cs:271]: You are very hilarious.

**Comment by hhyyrylainen** [comment]
I'm very serious. It's hard to find *everything* wrong with a PR in a single go, it's mentally taxing to review 4 PRs back to back, and also check that the changes get properly done.

**Review by hhyyrylainen — APPROVED** [review]

**Comment by Kemikal1** [comment]
Sure,i believe you.

**Pull request merged**